frontend-project/src/components/Table/DocumentTypeName.tsx

```diff
@@ -0,0 +1,16 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Spin } from 'antd';
+
+import { fetchDocumentType } from '@/services/documentType';
+
+export const DocumentTypeName: FC<{ id: number }> = props => {
+  const [name, setName] = useState('');
+  useEffect(() => {
+    const fetchData = async () => {
```

Comment:
po co ta funkcja jest tak zagnieżdżona? nie wystarczyłoby po prostu:
```
  useEffect(() => {
      const result = await fetchDocumentType(props.id);
      setName(result.name);
  }, []);

Reply:
Przy takiej konstrukcji pojawia się błąd: 'await' expressions are only allowed within async functions and at the top levels of modules.

Reply:
Próbowałem zrobić coś takiego:
```javascript
  useEffect(async () => {
      const result = await fetchDocumentType(props.id);
      setName(result.name);
  }, []);
```
i to wyglądało sensownie, nawet się bundlowału w webpacku, ale już podczas uruchomienia dostawałem:
```
index.js:1 Warning: An effect function must not return anything besides a function, which is used for clean-up.

It looks like you wrote useEffect(async () => ...) or returned a Promise. Instead, write the async function inside your effect and call it immediately:

useEffect(() => {
  async function fetchData() {
    // You can await here
    const response = await MyAPI.getData(someId);
    // ...
  }
  fetchData();
}, [someId]); // Or [] if effect doesn't need props or state
```

Reply:
Może to nie wygląda zbyt ładnie ale IMO dosyć popularne w podejściu funkcyjnym

useEffect(() => {(async () => {właściwa funkcja asynchroniczna})()})

Piszę z telefonu ale mam nadzieję, że widać o co chodzi
Po prostu funkcja anonimowa z natychmiastowym wywołaniem 
(async () => {treść})()